pkg/granted/registry/cfregistry/cfregistry.go

```diff
@@ -0,0 +1,93 @@
+package cfregistry
+
+import (
+	"context"
+	"fmt"
+
+	"connectrpc.com/connect"
+	"github.com/common-fate/sdk/config"
+	awsv1alpha1 "github.com/common-fate/sdk/gen/granted/registry/aws/v1alpha1"
+	"github.com/common-fate/sdk/gen/granted/registry/aws/v1alpha1/awsv1alpha1connect"
+	grantedv1alpha1 "github.com/common-fate/sdk/service/granted/registry"
+	"gopkg.in/ini.v1"
+)
+
+type Registry struct {
+	opts   Opts
+	Client awsv1alpha1connect.ProfileRegistryServiceClient
+}
+
+type Opts struct {
+	Name string
+	URL  string
+}
+
+func New(opts Opts) (*Registry, error) {
+
+	cfg, err := config.LoadDefault(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.APIURL != opts.URL {
+		return nil, fmt.Errorf("passed url does not match url in Common Fate (cf) config. active context API URL is: %s", cfg.APIURL)
+	}
```

Comment:
Let's remove this check. One of the goals of implementing this in Granted is that teams should be able to implement their own HTTP-based registry, by implementing the `granted.registry.aws.v1alpha1` protobuf API (for example, a team could write a custom Lambda function to serve AWS profiles). 

This check forces people to use the Common Fate API, which shouldn't be required for this feature

```suggestion
```

Reply:
@chrnorm Ahh yep this was different to my understanding. In that case we will need to handle API authentication better here too. 

At the moment it will rely on the Granted user having set up cf and having context already existing in the config and pulling from that to successfully authenticate with the CF API endpoint.
In our case we are setting up a client handler which calls Common Fate with the oidc auth. Same as how the `cf` cli would.  

Another case which you have mentioned is via a lambda call. This could be served directly via the lambda or via an api gateway, either way the auth here would be with AWS IAM. 

Either way the logic for handling this will differ for every case that an end user will need to use. Unless the API has no auth attached. Is the solution here to make the http registry generic enough to just handle a bearer token to be attached to an api call, leaving adding the logic for obtaining that auth token to the end user?

Reply:
I am thinking here for the initial implementation let's leave the existing auth as-is using OIDC. Because we use OIDC with Common Fate, if a user wanted to set up a custom Granted Profile Registry service they could also use OIDC to authenticate.

In that case the user would *still* use the `~/.cf/config` file to specify the OIDC issuer, even if they set up a custom HTTP registry.